test/unit/effects.js

```diff
@@ -1850,6 +1852,44 @@ test("Animation callbacks (#11797)", 15, function() {
 	this.clock.tick( 10 );
 });
 
+test( "Animation callbacks in order (#2292)", 10, function( assert ) {
+	var step = 0,
+		dur = 50;
+
+	// assert? -> github.com/JamesMGreene/qunit-assert-step
+	jQuery( "#foo" ).animate( {
+		width: "5px"
+	}, {
+		duration: dur,
+		start: function() {
+			assert.step( 1 );
+		},
+		progress: function( anim, p, ms ) {
+			if ( !( step++ ) ) {
+				assert.step( 2 );
+				equal( p,    0, "first call of progress: percent != 0" );
+				equal( ms, dur, "first call of progress: ms != duration (" + dur + ")" );
+			} else {
+				assert.step( 3 );
+				equal( p,  1, "last call of progress: percent != 1" );
+				equal( ms, 0, "last call of progress: ms != 0" );
```

Comment:
These `equal`s should be `strictEqual`s, and the messages ought to be more clear (e.g., `"(first|last) progress callback: (progress ratio|remaining ms)"`).